src/main/java/com/ecommerce/wlm_ecommerce_product_service/application/service/PaymentService.java

```diff
@@ -0,0 +1,37 @@
+package com.ecommerce.wlm_ecommerce_product_service.application.service;
+
+import com.ecommerce.wlm_ecommerce_product_service.application.dto.PaymentRequestDTO;
+import com.ecommerce.wlm_ecommerce_product_service.domain.exception.OrderAlreadyPaidException;
+import com.ecommerce.wlm_ecommerce_product_service.domain.exception.OrderNotFoundException;
+import com.ecommerce.wlm_ecommerce_product_service.domain.model.StatusPayment;
+import com.ecommerce.wlm_ecommerce_product_service.infrastructure.messaging.PaymentRequestProducer;
+import com.ecommerce.wlm_ecommerce_product_service.infrastructure.persistence.entity.OrderEntity;
+import com.ecommerce.wlm_ecommerce_product_service.infrastructure.persistence.repository.SpringDataOrder;
+import jakarta.validation.constraints.Positive;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class PaymentService {
+    private final PaymentRequestProducer paymentRequestProducer;
+    private final SpringDataOrder springDataOrder;
+
+    public void doPayment(@Positive Long orderId) throws OrderNotFoundException, OrderAlreadyPaidException {
+        OrderEntity order = springDataOrder.findById(orderId).orElseThrow(
+                () -> new OrderNotFoundException("Could not found order for this id"));
+
+        if(order.getStatusPayment().equals(StatusPayment.SUCCESS)){
```

Comment:
potentially order.getStatusPayment() can be null, however StatusPayment.SUCCESS is never null. So the better solution is  StatusPayment.SUCCESS.equals(order.getStatusPayment());